File: Data Structures & Algorithms/maximum-subarray/submission-2.java
class Solution {
    public int maxSubArray(int[] nums) {
        // Kadane's Algorithm
        // Compare two max of variables

        int currMax = 0;
        int resMax = 0;

        for(int i = 0; i < nums.length; i++){
            currMax = Math.max(0, nums[i] + currMax);
            resMax = Math.max(currMax, resMax);
        }

        return currMax <= 0 ? -1 : resMax;
    }
}

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-2.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        res = [0] * n

        for L in range(n - 1):
            curr_max = 0
            for R in range(L + 1, n):
                curr_max = max(curr_max, arr[R])

            res[L] = curr_max
        res[n-1] = -1
        return res